### ISSUE TITLE
I can't set iat claim as nil

### ISSUE THREAD

**Issue by sanrodari** [ueberauth/guardian#175]
Reproduction steps:

```elixir
claims = Guardian.Claims.app_claims |> Map.put("iat", nil)
Guardian.encode_and_sign(admin, :token, claims)

** (UndefinedFunctionError) undefined function Map.put_new/2
    (elixir) Map.put_new(%{"exp" => 1470582692, "iat" => nil, "iss" => "Zupport", "jti" => "0b568254-3888-4fd6-8da0-6ded7a796ec1", "pem" => %{}, "sub" => "Admin:1", "typ" => "token"}, 2467990693)
             lib/guardian.ex:371: Guardian.set_ttl/1
             lib/guardian.ex:345: Guardian.build_claims/3
             lib/guardian.ex:78: Guardian.encode_and_sign/3
```

**Comment by doomspork** [comment]
I believe this is fixed via #173, could you confirm @sanrodari?

**Comment by sanrodari** [comment]
Effectively @doomspork! 😃 

mix.exs
```elixir
...
{:guardian, github: "nimish-mehta/guardian", ref: "fix-assign-exp"},
...
```


iex
```elixir
claims = Guardian.Claims.app_claims |> Map.put("iat", nil)
Guardian.encode_and_sign(account, :token, claims)
{:ok,
 "eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJBY2NvdW50OjEiLCJleHAiOjI0NjgwMTUwODUsImlhdCI6bnVsbCwiaXNzIjoiUG90aW9uIiwianRpIjoiZWIyYmMzNDEtNDI0Mi00N2MxLTk5OTQtNDQ0N2I4ZDQ2NjBiIiwicGVtIjp7fSwic3ViIjoiQWNjb3VudDoxIiwidHlwIjoidG9rZW4ifQ.j-kGU82tmoG9HyaLToNUWVb-ER8IvtsOeiIV_2eBFy70mYosCTz1vp7Xb3b4Z5CRum5m7vyvwN--oFOdU82LTA",
 %{"aud" => "Account:1", "exp" => 2468015085, "iat" => nil, "iss" => "Potion",
   "jti" => "eb2bc341-4242-47c1-9994-4447b8d4660b", "pem" => %{},
   "sub" => "Account:1", "typ" => "token"}}
```

**Comment by doomspork** [comment]
This was actually fixed by #169, we'll release a new version shortly.
